Server/src/main/webapp/resources/js/model.js

```diff
@@ -0,0 +1,145 @@
+class Model {
+    _posts;
+
+    constructor() {
+        this._posts = [];
+
+        let values = [];
+        let keys = Object.keys(localStorage);
+        let i = keys.length;
+
+        while ( i-- ) {
+            if(Number.parseInt(keys[i])) {
+                let value = JSON.parse(localStorage.getItem(keys[i]));
+                value.createdAt = new Date(value.createdAt);
+                values.push(value);
+            }
+        }
+
+        this._posts = values.sort((a, b) => a.id - b.id);
+    }
+
+    _postSchema = {
+        id: val => typeof val === 'number',
+        description: val => typeof val === 'string' && val.length < 200,
+        createdAt: val => Object.prototype.toString.call(val) === '[object Date]',
+        author: val => typeof val === 'string' && val.length > 0,
+        photoLink: val => typeof val === 'string',
+        hashTags: val => typeof val === 'string' && val.length > 0 && val.includes('#'),
+        likes: val => Array.isArray(val)
+    };
+
+    validatePost(post) {
+        if (this._posts.find(p => post.id === p.id) != null) {
+            console.log('this id already exist');
+            return false;
+        }
+
+        if ((Math.abs(Object.keys(post).length - Object.keys(this._postSchema).length) === 1 && post['photoLink'] != null)
+            || Math.abs(Object.keys(post).length - Object.keys(this._postSchema).length) > 1) {
+            console.log('illegal arguments number');
+            return false;
+        }
+
+        return this._validateForm(post);
+    }
+
+    _validateForm(form) {
+        let errors = Object.keys(form)
+            .filter(key => !this._postSchema[key]?.(form[key]))
+            .map(key => new Error(key + ' is invalid!'));
+
+        if (errors.length > 0) {
+            alert(errors.map(error => error.message).join(" "));
+            return false;
+        }
+
+        return true;
+    }
+
+    addPost(post) {
+        if (this._posts.length === 0) {
+            post.id = 1;
+        } else {
+            post.id = this._posts[this._posts.length - 1].id + 1;
+        }
```

Comment:
`post.id = this._posts.length === 0 ? 1 : this._posts[this._posts.length - 1].id + 1;`